hydroflow/tests/groupby.rs

```diff
@@ -0,0 +1,112 @@
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::rc::Rc;
+
+use hydroflow::lang::collections::Iter;
+use hydroflow::scheduled::graph::Hydroflow;
+use hydroflow::scheduled::graph_ext::GraphExt;
+use hydroflow::scheduled::handoff::VecHandoff;
+
+const BATCH_A: [&'static str; 7] = ["megan", "davis", "mingwei", "john", "justin", "joe", "mae"];
+const BATCH_B: [&'static str; 7] = [
+    "mingwei", "lauren", "justin", "mae", "mingwei", "justin", "pierce",
+];
+const BATCH_C: [&'static str; 2] = ["joe", "mae"];
+
+#[test]
+fn groupby_core_monotonic() {
```

Comment:
I think this test could use a brief explanation of what it's doing (it seems like: "release each value once its been seen three times"?)